cpp/CMakeLists.txt

```diff
@@ -463,7 +455,7 @@ if(NOT BUILD_CPU_ONLY)
   target_link_libraries(
     cuvs
     PUBLIC rmm::rmm raft::raft ${CUVS_CTK_MATH_DEPENDENCIES}
-    PRIVATE nvidia::cutlass::cutlass $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>
+    PRIVATE nvidia::cutlass::cutlass $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX> cuvs-cagra-search
```

Comment:
Was building a new artifact really necessary to improve the perf / binary size? Or was this just done to make the build more modular?

Reply:
Generally, separable compilation affects performance negatively, so I reduced the AOE by setting cmake CUDA_SEPARABLE_COMPILATION on the relevant files only - via this component.